### PR TITLE
PE-1663: Feedback Modal triggers

### PR DESCRIPTION
**Pull request by matibat**
The modal has to appear no more than once per session.
The events that will trigger the modal to open will be only:
- Upload folder
- Upload file
- Manifest Creation
- Sharing drive/file

### CODE REVIEW

File: lib/blocs/feedback_survey/feedback_survey_cubit.dart [hunk 1]
```diff
@@ -4,19 +4,19 @@ import 'package:flutter_bloc/flutter_bloc.dart';
...
+  bool hasAlreadyBeenOpened = false;
+
   FeedbackSurveyCubit(FeedbackSurveyState initialState) : super(initialState);
-  bool hasAlreadyBeenOpenedFromSync = false;
 
-  void openRemindMe(String source) {
-    if (!(hasAlreadyBeenOpenedFromSync && source == 'sync')) {
-      emit(FeedbackSurveyRemindMe(isOpen: true, source: source));
+  void openRemindMe() {
+    if (!(hasAlreadyBeenOpened)) {
```
Comment (matibat) [inline lib/blocs/feedback_survey/feedback_survey_cubit.dart:12]: This ensures we open the modal no more than once per session

File: lib/components/create_manifest_form.dart [hunk 2]
```diff
@@ -15,6 +15,7 @@ import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:reactive_forms/reactive_forms.dart';
 import 'package:url_launcher/url_launcher.dart';
 
+import '../blocs/feedback_survey/feedback_survey_cubit.dart';
```
Comment (matibat) [inline lib/components/create_manifest_form.dart:18]: We should use 'package:...'

File: lib/components/create_manifest_form.dart [hunk 3]
```diff
@@ -52,6 +53,7 @@ class CreateManifestForm extends StatelessWidget {
             state is CreateManifestPrivacyMismatch) {
           Navigator.pop(context);
           Navigator.pop(context);
+          context.read<FeedbackSurveyCubit>().openRemindMe();
```
Comment (matibat) [inline lib/components/create_manifest_form.dart:56]: This triggers the modal on manifest creation success

File: lib/components/drive_share_dialog.dart [hunk 4]
```diff
@@ -6,6 +6,7 @@ import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
+import '../blocs/feedback_survey/feedback_survey_cubit.dart';
```
Comment (matibat) [inline lib/components/drive_share_dialog.dart:9]: We should use 'package:...'

File: lib/components/drive_share_dialog.dart [hunk 5]
```diff
@@ -97,7 +98,10 @@ class _DriveShareDialogState extends State<DriveShareDialog> {
           actions: [
             if (state is DriveShareLoadSuccess)
               ElevatedButton(
-                onPressed: () => Navigator.pop(context),
+                onPressed: () {
+                  Navigator.pop(context);
+                  context.read<FeedbackSurveyCubit>().openRemindMe();
```
Comment (matibat) [inline lib/components/drive_share_dialog.dart:103]: This triggers the modal after closing the drive share modal

File: lib/components/file_share_dialog.dart [hunk 6]
```diff
@@ -98,7 +99,10 @@ class _FileShareDialogState extends State<FileShareDialog> {
           actions: [
             if (state is FileShareLoadSuccess)
               ElevatedButton(
-                onPressed: () => Navigator.pop(context),
+                onPressed: () {
+                  Navigator.pop(context);
+                  context.read<FeedbackSurveyCubit>().openRemindMe();
```
Comment (matibat) [inline lib/components/file_share_dialog.dart:104]: This triggers the modal after closing the file share modal

File: lib/components/upload_form.dart [hunk 7]
```diff
@@ -13,6 +13,7 @@ import 'package:ardrive/utils/upload_plan_utils.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
+import '../blocs/feedback_survey/feedback_survey_cubit.dart';
```
Comment (matibat) [inline lib/components/upload_form.dart:16]: We should use 'package:...'

File: lib/components/upload_form.dart [hunk 8]
```diff
@@ -78,6 +79,7 @@ class UploadForm extends StatelessWidget {
         listener: (context, state) async {
           if (state is UploadComplete || state is UploadWalletMismatch) {
             Navigator.pop(context);
+            context.read<FeedbackSurveyCubit>().openRemindMe();
```
Comment (matibat) [inline lib/components/upload_form.dart:82]: This triggers the modal on upload success

File: lib/pages/app_router_delegate.dart [hunk 9]
```diff
@@ -132,43 +132,46 @@ class AppRouterDelegate extends RouterDelegate<AppRoutePath>
                     driveDao: context.read<DriveDao>(),
                     config: context.read<AppConfig>(),
                   ),
-                  child: BlocListener<DriveDetailCubit, DriveDetailState>(
```
Comment (matibat) [inline lib/pages/app_router_delegate.dart:135]: The changes here only replaces the nested `BlocListener`s with `MultiBlocListener`

File: lib/components/upload_form.dart (hunk 7 above)
Comment (matibat) [reply lib/components/upload_form.dart]: 14ba2a2228575f6562254abf9b0c516fe7d9842f

File: lib/components/drive_share_dialog.dart (hunk 4 above)
Comment (matibat) [reply lib/components/drive_share_dialog.dart]: 14ba2a2228575f6562254abf9b0c516fe7d9842f

File: lib/components/create_manifest_form.dart (hunk 2 above)
Comment (matibat) [reply lib/components/create_manifest_form.dart]: 14ba2a2228575f6562254abf9b0c516fe7d9842f

**Review by karlprieb — APPROVED** [review]

**Pull request merged**